### ISSUE TITLE
can make slug from 2 or more fields ?

### ISSUE THREAD

**Issue by lyquocnam** [jagi/meteor-astronomy-slug-behavior#9]
can make slug from 2 or more fields ?
ex:
i have some fields: code, name, price, slug...
i want make slug from: name + code + year...
this can be ?

**Comment by lukejagodzinski** [comment]
Yes, you can. Take a look at the "The fieldName and methodName" section here https://github.com/jagi/meteor-astronomy-docs/blob/master/docs/slug.md You have to provide the `methodName` option (function) instead `fieldName`.

**Comment by lyquocnam** [comment]
i'm done with tutorial, but slug field null. @jagi 

my class:

```
import {Astro} from 'meteor/jagi:astronomy';
import {Validators} from 'meteor/jagi:astronomy-validators';
import {timestamp} from 'meteor/jagi:astronomy-timestamp-behavior';
import 'meteor/jagi:astronomy-slug-behavior';
import {relations} from 'meteor/jagi:astronomy-relations';
import {Products, Categories} from '/imports/lib/collections';
//import {formatNumber} from '/client/lib/formater';

export default Astro.Class({
  name: 'Product',
  collection: Products,
    behaviors: {
      timestamp: {
        hasCreatedField: true,
        createdFieldName: 'createdAt',
        hasUpdatedField: true,
        updatedFieldName: 'updatedAt'
      },
        slug: {
        fieldName: null,
      methodName: 'makeSlug',
        //slugFieldName: 'slug',
        //canUpdate: true,
        //unique: true,
        separator: '-'
      }
    },
  fields: {
        code: 'string',
        name: 'string',
        slug: 'string',
    price: 'number',
    unit: 'string',
    photo: 'string',
    inStock: {
      type: 'boolean',
      default: true,
    },
    body: 'string',
    cateId: {
      type: 'string',
      default: Categories.findOne() ? Categories.findOne()._id : null,
    },
    // category: {
        //  type: 'object',
        //  nested: 'Category',
    //   optional: true,
        // },
    },
  relations: {
   category: {
     type: 'one',
     class: 'Category',
     local: 'cateId',
     foreign: '_id'
   }

 },
    validators: {
    code: Validators.and([
      Validators.required(),
    ]),
    name: Validators.and([
      Validators.required(),
    ]),
    cateId: Validators.and([
      Validators.required(),
    ]),
  },
    events: {

    validationError: function(e) {
            if (e.data.validator.name == 'required' && e.data.fieldName == 'code') {
                e.setMessage('Bạn chưa nhập [Mã sản phẩm].');
                //e.stopPropagation();
            }
      if (e.data.validator.name == 'required' && e.data.fieldName == 'name') {
                e.setMessage('Bạn chưa nhập [Tên sản phẩm].');
                //e.stopPropagation();
            }
      if (e.data.validator.name == 'required' && e.data.fieldName == 'cateId') {
                e.setMessage('Bạn chưa chọn [Nhóm sản phẩm].');
                //e.stopPropagation();
            }
      e.stopPropagation();
        }

    },

  methods: {
    makeSlug() {
      return this.name + ' ' + this.code;
    },
    getCategoryName(){
      const category = this.category();
      return category ? category.name : '';
    }
  },
});

```

optional idea: can we only pass object type for `fieldName`:
`fieldName: ['name','code']`
this may simple

**Comment by lukejagodzinski** [comment]
Oh ok in deed there is a bug. I will try to fix it today.

**Comment by lukejagodzinski** [comment]
Bug fixed in `jagi:astronomy-slug-behavior@1.3.1`
